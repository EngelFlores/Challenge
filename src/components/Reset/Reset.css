.reset-button {
  border: 0;
  width: 100%;
  background-color: white;
  height: 40px;
  font-size: 15px;
  border-radius: 20px;
  min-width: min-content;
}