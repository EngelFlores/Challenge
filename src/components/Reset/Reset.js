import React from 'react';
import "./Reset.css"

const Reset = ({ handleResetFilters }) => {
  return (
    <button className="reset-button" data-testid="reset-button" onClick={handleResetFilters}>Reset all filter</button>
  )
}

export { Reset }

