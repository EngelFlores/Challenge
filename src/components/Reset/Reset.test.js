import { render, waitFor, fireEvent } from '@testing-library/react';
import React from 'react'

import { Reset } from './Reset'

describe('Should render reset button', () => {
  test('Render filtered information from user', async () => {
    const { getByTestId } = render(<Reset handleResetFilters={() => false} />)
    const resetButton = await waitFor(() => getByTestId('reset-button'))
    fireEvent.click(resetButton)

    expect(resetButton).toBeInTheDocument()
  })

  test('Component match snapshot', async () => {
    const { getByTestId } = render(<Reset handleFilter={() => false} />)
    const button = await waitFor(() => getByTestId('reset-button'))

    expect(button).toMatchSnapshot()
  })

})