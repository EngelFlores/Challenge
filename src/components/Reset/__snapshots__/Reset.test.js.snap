// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render reset button Component match snapshot 1`] = `
<button
  class="reset-button"
  data-testid="reset-button"
>
  Reset all filter
</button>
`;
